clang/include/clang/Tooling/DependencyScanning/DependencyScanningService.h

```diff
@@ -95,6 +96,10 @@ class DependencyScanningService {
     return SharedCache;
   }
 
+  ModuleCacheMutexes &getSharedModuleCacheMutexes() {
```

Comment:
Not sure whether the service should hand out `ModuleCacheMutexes &` (the same thing we do for the shared caching filesystem) or whether it'd be simples to just hand out ready thread-safe `std::shared_ptr<ModuleCacheLock>`. Thoughts?

Reply:
I don't think we need shared_ptr here, the lifetime of workers is already bound to the service.

Reply:
We don't _need_ it, but we currently call `std::make_shared<ModuleCacheMutexLock>` for each TU, which is unnecessary. We could just do that on service construction and avoid future allocations.